tools/cmake/toolchains/GCC_ARM.cmake

```diff
@@ -67,23 +67,23 @@ function(mbed_set_toolchain_options target)
 endfunction()
 
 # GCC ARM requires preprecessing linker script, execute generators to get definitions needed for
-# this step - linker options and compile definitions 
-function(mbed_generate_gcc_options_for_linker target definitions_file linker_options_file)
+# this step - compile definitions
+function(mbed_generate_gcc_options_for_linker target definitions_file)
     set(_compile_definitions 
         "$<TARGET_PROPERTY:${target},COMPILE_DEFINITIONS>"
     )
-
-    set(_linker_options
-        "$<TARGET_PROPERTY:${target},LINK_OPTIONS>"
+    # We filter -D to the linker script, as only simple ones are passed (value based)
+    # To properly fix, we will need to replace all these the same way as Ninja or other
+    # generator do
+    set(_compile_definitions
+       "$<FILTER:${_compile_definitions},EXCLUDE,.*[ ={ | =" | =\( ].*[ } | " | )]>"
     )
 
     set(_compile_definitions 
```

Comment:
Please add a comment here to explain what it is doing.

Reply:
fixed